asn1/CommonLedgerProtocol.asn

```diff
@@ -21,81 +21,53 @@ SideProtocolData ::= SEQUENCE OF SEQUENCE {
   protocolData OCTET STRING
 }
 
-PacketRequest ::= SEQUENCE {
-  packet InterledgerPacket,
+Ack ::= SEQUENCE {
+  ack NULL,
```

Comment:
We seem to be mixing layers a bit here. `Prepare`, `Fulfill` and `Reject` are logical operations. They relate to a transfer not a request/response pair, that should give us a clue. We should define their semantics separately from the semantics of matching requests and responses.

Sending a message that doesn't need a response isn't something you should have to define at the logical layer (and at the messaging layer, I'd argue it doesn't matter). If I send a request for which I don't need a response why does the server care?

If the logical interaction is async then I should get back an Ack response at the messaging layer (i.e. A request/response pair that only cover the logical request).

Then the async response can be covered in a new request/response pair at the messaging layer (initiated in the opposite direction).

Using a requestId of zero seems to have no obvious benefit it just raises a lot of questions for me:
- Are we expecting a server to not know if it's operating in an async or synchronous mode?
- Are we expecting the receiver to not understand the logical request and so not know if it must ack or respond?
- Are we expecting a client to throw an error if it gets an Ack and doesn't want it?
- How do I track down a request from the logs if all notifications have the same id (zero)?